### PR TITLE
Rearrange CLI modules, part 2

### PR DESCRIPTION
**Pull request by dcoutts**
Rearrange the Byron CLI modules to try and follow a similar structure
to that we have adopted for the Shelley CLI modules.

This clarifies somewhat what is Byron-specific vs what is shared/general.

### CODE REVIEW

File: cardano-cli/src/Cardano/CLI/Byron/Delegation.hs [hunk 1]
```diff
@@ -20,7 +20,7 @@ import           Cardano.Config.Types (CertificateFile (..))
 import qualified Cardano.Crypto as Crypto
 import           Cardano.Crypto (ProtocolMagicId)
 
-import           Cardano.CLI.Ops
+import           Cardano.CLI.Errors (CliError(..))
```
Comment (erikd) [inline cardano-cli/src/Cardano/CLI/Byron/Delegation.hs:23]: :100: This has been annoying me for the longest time but was worried about breaking WIP by moving it.

File: cardano-cli/src/Cardano/CLI/Parsers.hs [hunk 2]
```diff
@@ -1,242 +1,35 @@
...
-      "Show cardano-cli version"
-      $ pure DisplayVersion
-  , command'
-      "pretty-print-cbor"
-      "Pretty print a CBOR file."
-      $ PrettyPrintCBOR
-          <$> parseFilePath "filepath" "Filepath of CBOR file."
-  ]
-
+-}
```
Comment (erikd) [inline cardano-cli/src/Cardano/CLI/Parsers.hs]: ?? Is that some code that can just be dropped?

File: cardano-cli/src/Cardano/CLI/Run.hs [hunk 3]
```diff
@@ -1,93 +1,31 @@
-{-# LANGUAGE GADTs #-}
-{-# LANGUAGE NamedFieldPuns #-}
-{-# LANGUAGE OverloadedStrings #-}
-{-# LANGUAGE ScopedTypeVariables #-}
-{-# LANGUAGE StrictData #-}
-
-{-# OPTIONS_GHC -Wno-all-missed-specialisations #-}
```
Comment (erikd) [inline cardano-cli/src/Cardano/CLI/Run.hs:7]: :100:

File: cardano-cli/cardano-cli.cabal [hunk 4]
```diff
@@ -108,6 +108,7 @@ library
                        -Wredundant-constraints
                        -Wpartial-fields
                        -Wcompat
+                       -Wno-all-missed-specialisations
```
Comment (erikd) [inline cardano-cli/cardano-cli.cabal:116]: Do we even need that anywhere? We did when we were using `-Weverything` but we dropped than in favour of `-Wall` some time ago. Also, doesn't `cabal build` from the top level ignore these warning flags in favour of the ones in `cabal.project`

File: cardano-cli/src/Cardano/CLI/Byron/Key.hs [hunk 5]
```diff
@@ -1,13 +1,7 @@
-{-# LANGUAGE CPP #-}
 {-# LANGUAGE GeneralisedNewtypeDeriving #-}
-{-# LANGUAGE LambdaCase #-}
-{-# LANGUAGE NoMonomorphismRestriction #-}
 
-{-# OPTIONS_GHC -Wno-all-missed-specialisations #-}
 {-# OPTIONS_GHC -Wno-missing-local-signatures #-}
```
Comment (erikd) [inline cardano-cli/src/Cardano/CLI/Byron/Key.hs:3]: That would be a good one to remove.

File: cardano-cli/src/Cardano/CLI/Parsers.hs (hunk 2 above)
Comment (dcoutts) [reply cardano-cli/src/Cardano/CLI/Parsers.hs]: It gets re-enabled in the next patch.

File: cardano-cli/cardano-cli.cabal (hunk 4 above)
Comment (dcoutts) [reply cardano-cli/cardano-cli.cabal:116]: I don't know if we do need it. cabal does not ignore anything, that's stack.

File: cardano-cli/src/Cardano/CLI/Byron/Key.hs (hunk 5 above)
Comment (dcoutts) [reply cardano-cli/src/Cardano/CLI/Byron/Key.hs:3]: We have to fix the monolithic error types first :-)

**Comment by dcoutts** [comment]
bors merge

**Pull request merged**